CashOverflow.Tests.Unit/Services/Foundations/Languages/LanguageServiceTests.Exceptions.Add.cs

```diff
@@ -114,5 +115,84 @@ public async Task ShouldThrowServiceExceptionOnAddIfServiceErrorOccursAndLogItAs
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Throws(dbUpdateConcurrencyException);
+
+            // when
+            ValueTask<Language> addLanguageTask = this.languageService.AddLanguageAsync(someLanguage);
+
+            LanguageDependencyValidationException actualLanguageDependencyValidationException =
+                 await Assert.ThrowsAsync<LanguageDependencyValidationException>(addLanguageTask.AsTask);
+
+            // then
+            actualLanguageDependencyValidationException.Should()
+                .BeEquivalentTo(expectedLanguageDependencyValidationException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker => broker.LogError(It.Is(
+                SameExceptionAs(expectedLanguageDependencyValidationException))), Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertLanguageAsync(It.IsAny<Language>()), Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnAddIfDbUpdateExceptionOccuredAndLogItAsync()
+        {
+            // given
+            Language someLanguage = CreateRandomLanguage();
+            string someMessage = GetRandomString();
+
+            var dbUpdateException = new DbUpdateException(someMessage);
+
+            var failedLanguageStorageException =
+                new FailedLanguageStorageException(dbUpdateException);
+
+            var expectedLanguageDependencyException =
+                new LanguageDependencyException(failedLanguageStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                    broker.GetCurrentDateTimeOffset()).Throws(dbUpdateException);
+
+            // when
+            ValueTask<Language> addLanguageTask =
+                this.languageService.AddLanguageAsync(someLanguage);
+
+            LanguageDependencyException actualLanguageDependencyException =
+                await Assert.ThrowsAsync<LanguageDependencyException>(addLanguageTask.AsTask);
+
+            // then
+            actualLanguageDependencyException.Should().BeEquivalentTo(expectedLanguageDependencyException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLanguageDependencyException))), Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```

Comment:
Bu yerdagi ketma-ketlik to'g'rilanishi kerak. Logika bo'yicha datetime, logging chaqariladi va strorage chaqirilmaydi shuning uchun
datetime
logging
storage 

ketma-ket bo'lishi kerak

Reply:
Done